include/bluetooth/enocean.rst

```diff
@@ -0,0 +1,112 @@
+However, when using the library with your application, you can configure the limit of supported devices to a higher number.
+
+Usage
+=====
+
+The EnOcean library uses :cpp:func:`zephyr:bt_le_scan_cb_register` to receive Bluetooth advertisement packets from nearby EnOcean devices.
+The Bluetooth stack must be initialized separately, and Bluetooth scanning must be started by the application.
+The Bluetooth Mesh stack does this automatically, so when combined, the application does not need to perform any additional initialization.
+
+.. _bt_enocean_commissioning:
+
+Commissioning
+*************
+
+Before the library is able to process sensor or button messages from nearby EnOcean devices, the devices must be commissioned.
+By default, the commissioning behavior may be triggered by interacting with the devices, as explained in the following sections.
+
+To enable radio-based commissioning in your application, call :cpp:func:`bt_enocean_commissioning_enable`.
+
+Commissioning wall switches
+---------------------------
+
+EnOcean Easyfit wall switches may be put into the commissioning mode by completing the following sequence:
+
+1. Press and hold one of the buttons for more than 7 seconds before releasing it.
+#. Quickly press and release the same button (hold for less than 2 seconds).
+#. Press and hold the same button again for more than 7 seconds before releasing it.
+
+The switch will transmit its security key, which the library picks up and stores before calling the :cpp:member:`bt_enocean_callbacks::commissioned` callback.
+
+To exit the commissioning mode on a wall switch device, press any of the other buttons on the switch.
+The library will now be reporting button presses through the :cpp:member:`bt_enocean_callbacks::button`.
+
+Commissioning sensor units
+--------------------------
+
+To have the EnOcean Easyfit sensor unit modules publish their security key, short-press the front button of the device.
+The library registers the security key and stores it before calling :cpp:member:`bt_enocean_callbacks::commissioned`.
+
+The sensor unit never enters the commissioning mode.
+It just sends commissioning messages on the button press.
+
+.. note::
+   EnOcean devices may be configured to disable radio-based commissioning through NFC.
+   In such case, the security key must be obtained through manual input.
+
+Observing output
+****************
+
+To observe the output of the EnOcean devices, integrate the library with your application.
+See the :ref:`enocean_sample` sample for an example of such integration that uses an nRF board for observation.
+
+.. enocean_observing_output_start
+
+The observer device forwards incoming advertisements to the EnOcean library for processing.
+The application receives events from the EnOcean library through callbacks, and prints the outcome to console.
+The LEDs of the board also respond to button presses from an EnOcean switch.
```

Comment:
This describes the sample, not the library. Users of the library do not have to print to console or change LEDs. I don't think this section is required at all, really, but if we need to have something for the post-commissioning phase, I suggest just putting the following sentence:

> After commissioning an EnOcean device, its activity may be monitored through the :ref:`bt_enocean_handlers` callback functions passed to :ref:`bt_enocean_init`. See the :ref:`enocean_sample` for a demonstration of the handler callback functions.